util/http/retry.go

```diff
@@ -27,13 +28,101 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-// defaults based on a target total retry time of at least 5s. 30*((2^8)-1)>5000
 const (
-	DefaultMaxRetries        = 8
-	DefaultMinWaitMsec int64 = 30
-	DefaultMaxWaitMsec int64 = 300000
+	// DefaultDialTimeoutMsec is the default number of milliseconds before timeout while connecting to a remote endpoint. See `TimeoutConfig.DialTimeout`.
+	DefaultDialTimeoutMsec = 3000
+	// DefaultResponseHeaderTimeoutMsec is the default number of milliseconds before timeout while waiting for response header from a remote endpoint. See `TimeoutConfig.ResponseHeaderTimeout`.
+	DefaultResponseHeaderTimeoutMsec = 3000
+	// DefaultRequestTimeoutMsec is the default number of milliseconds that the entire request can take before timeout. See `TimeoutConfig.RequestTimeout`.
+	DefaultRequestTimeoutMsec = 30_000
```

Comment:
Where was this one used previously?

Reply:
This was the default overall `http.Timeout` defined here https://pkg.go.dev/net/http#DefaultTransport

Reply:
Yeah, just making the overall default timeout explicit.